### ISSUE TITLE
Press clean on GUI should has confirmation

### ISSUE THREAD

**Issue by apinprastya** [kamilion/customizer#178]
I just accidentally press clean. And all of my work gone. It should be confirmation dialog when pressed.

**Comment by kamilion** [comment]
Okay, I don't know how to define new dialogs in QT's designer; if you can submit a patch to gui.ui, I can review and merge it. 

@clearkimura was the last one to touch gui.ui, removing a bunch from the about tab in this commit:
https://github.com/kamilion/customizer/commit/2d7b8f262f63f53db80338b326f8b2d6d1adde7d

Otherwise, there's plenty of other ways to shoot yourself in the foot during ISO creation, I'd suggest creating some scripts to perform the build for you, as I have.
https://github.com/kamilion/kamikazi-core/tree/master/buildscripts/xenial

Everything I've made in that repo is under the WTFPL, so feel free to fork, reuse, remix, or whatever.

**Comment by apinprastya** [comment]
@kamilion : if on Qt C++ there is a QMessageDialog::Question to make a pop up confirmation dialog. There should be the same for PyQt, but unfortunately I never code on PyQt.
About the buildscript, it a good think when we already know what we want to do, but when we just start the remaster usually we just try to add and remove things, and check if it work or not.

**Comment by kamilion** [comment]
I'll look into QMessageDialog, thanks.

Took me a long time (months) to determine which packages could safely be removed from lubuntu ISOs without breaking the desktop tools.

Unneeded Desktop Apps:
https://github.com/kamilion/kamikazi-core/blob/master/buildscripts/xenial/02-purgelist.synpkg
Orphaned Application Libraries:
https://github.com/kamilion/kamikazi-core/blob/master/buildscripts/xenial/03-purgelist.synpkg
Multimedia Libraries (Potential security risks for "servers"):
https://github.com/kamilion/kamikazi-core/blob/master/buildscripts/xenial/04-purgelist.synpkg


Once I finished breaking things by testing package removals, figuring out what I wanted to add was a *lot* easier and took barely any time at all.

Using synaptic to export package modifications was probably one of the smartest choices I made.
(all the .synpkg files come from "Save selected packages" in synaptic's file menu.)

The second smartest choice, I think, was to do things in two steps... All the purge operations to get down to an up-to-date minimal ISO, followed by a second build _using the fresh minimal ISO as a source_ to build back up.

https://github.com/kamilion/kamikazi-core/blob/master/buildscripts/xenial/00-build-minilubuntu-iso-from-source.sh

When I start doing a lot of iterations, having a clean minimal speeds things up **immensely**.

Most of the later scripts in the chain are pulling development tools into the ISO and building a few things there aren't .deb packages for, and are mostly either ignorable or useful for study.

Hope this is useful advice to you. Will do what I can to add a safety net to the GUI.

**Comment by apinprastya** [comment]
Thank you for the advice, it is very useful especially about the purgelist. But something I don't know is even the ISO is minimal Lubuntu but the size is still quite big (~900MB).

**Comment by kamilion** [comment]
After the purgelists I have defined are run, I get a minimal lubuntu iso at 580MB.

![Image of Lubuntu Desktop](https://puu.sh/xar1v.png)

**Comment by kamilion** [comment]
Looks like it's not very easy to insert a confirmation dialog.

QMessageBox uses new-style signals to communicate, and I just *barely* managed to convert customizer to use new-style signals to flag the main thread to open QMessageBox.

https://github.com/kamilion/customizer/blob/master/src/gui.py.in#L57

This is probably a pretty hamfisted way of doing this, but it's the only way I could think of to get 2.7 and 3.3+ working from the same codebase.
https://github.com/kamilion/customizer/blob/master/src/gui.py.in#L87

As it is, I don't really know how to deal with whatever it is that a QMessageBox returns; I just assign to an unused variable.

https://github.com/kamilion/customizer/blob/master/src/gui.py.in#L120

QT and GUI stuff aren't really my forte, I inherited this project and I can kinda barely keep it maintained.
If I hadn't, it would have ended up like ubuntu-builder (the last remaster tool I kept alive, and a sibling of the pre-python Customizer, both using GAMBAS basic) and quietly faded off into obscurity.

But because @fluxer did such great work writing the initial python replacement for Customizer, it's been pretty easy to fix python version incompatibilities. He gets all credit for writing the code, I'm just keeping it on life support. :)
